### PR TITLE
Timer for Polling event

### LINKED ISSUE #453: Add a timer on the event
**User Story**
Det bør utvikles en funksjon som synliggjør for deltakere og møteansvarlige hvor mange som er innlogget på løsningen i appen (live), slik at ordstyrer har grunnlag for å si at majoriteten har votert.
**Additional Description**

**Acceptance Criteria**
1. add timer on top when a question is published
2. add participant count (of all possible participants)
3. Design:
https://www.figma.com/file/Eq4kNcMejzrIQWV6sD4e0C/01.00-BCC-Voting?type=design&node-id=2013-8319

4. correct design for timer: 
image

### PR DESCRIPTION
**Pull request by StevenMalaihollo**
1/2 of #453

### CODE REVIEW

File: client/src/components/timer.vue [hunk 1]
```diff
@@ -0,0 +1,62 @@
+<template>
+    <div class="flex gap-x-1 opacity-0 transition-all ease-in-out duration-300" :class="{'opacity-100': show}">
+        <div class="flex flex-col items-center gap-y-1">
+            <div class="timer-box">
+                <label> {{ elapsedMinutes }} </label>
```
Comment (JakubC-projects) [inline client/src/components/timer.vue:5]: I think minutes should also be padded to always be 2-digit

File: client/src/components/timer.vue [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            </div>
+            <p class="caption">Min</p>
+        </div>
+        <span class="mt-1">:</span>
+        <div class="flex flex-col items-center gap-y-1">
+            <div class="timer-box">
+                <label> {{ elapsedSeconds < 10 ? `0${elapsedSeconds}` : elapsedSeconds }} </label>
+            </div>
+            <p class="caption">Sec</p>
```
Comment (JakubC-projects) [inline client/src/components/timer.vue:14]: This should probably be translated

File: client/src/components/timer.vue [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            maxElapsedSeconds: 60 * 60,
+            elapsedMinutes: 0,
+            elapsedSeconds: 0,
+            show: false,
+        };
+    },
+    mounted() {
+        setInterval(() => {
+            this.updateElapsedTime();
+        }, 1000);
```
[the remark this answers is not in the document]
Comment (JakubC-projects) [reply client/src/components/timer.vue:35]: We may actually need to think about the fact that device clocks may not be synchronised.

[the remark this answers is not in the document]
Comment (JakubC-projects) [reply client/src/components/timer.vue:35]: But aside from that we should at least have the timer synced if the device clocks are synced
For that you could consider those solutions:
1. Reduce the interval time (if we set it to e.g. 50 ms, then maximum time out of sync will be 50 ms)
2. Make sure the interval fires at the same time for each user - so basically to make sure that the interval is fired on whole seconds, not in between seconds.

File: client/src/components/timer.vue [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    },
+    methods: {
+        updateElapsedTime() {
+            const currentTime = Date.now();
+            const elapsed = Math.floor((currentTime - this.sinceDate) / 1000);
+            if (elapsed >= this.maxElapsedSeconds) {
```
Comment (JakubC-projects) [inline client/src/components/timer.vue:41]: I"m not sure about this one, why does it need to disappear when it reaches 1 hour

File: client/src/components/timer.vue [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+<script lang="ts">
+import {defineComponent} from 'vue';
+export default defineComponent({
+    props: {
+        sinceDate: {type: Number, required: true},
+    },
+    data() {
+        return {
+            maxElapsedSeconds: 60 * 60,
+            elapsedMinutes: 0,
```
Comment (JakubC-projects) [inline client/src/components/timer.vue:27]: I would presonally only store the elapsedTime (in ms) in the data, and then have those minutes and seconds as calculated properties

File: client/src/components/timer.vue [hunk 6]
```diff
@@ -0,0 +1,62 @@
+<template>
+    <div class="flex gap-x-1 opacity-0 transition-all ease-in-out duration-300" :class="{'opacity-100': show}">
```
[the remark this answers is not in the document]
Comment (StevenMalaihollo) [reply client/src/components/timer.vue:2]: Oh, I wasn't here is no time to count down from defined either

File: client/src/components/timer.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (StevenMalaihollo) [reply client/src/components/timer.vue:35]: Decreased the interval to `100` ms, I think it being out of sync for just 100ms is no problem.

File: client/src/components/timer.vue (hunk 1 above)
Comment (StevenMalaihollo) [reply client/src/components/timer.vue:5]: Fixed

File: client/src/components/timer.vue (hunk 4 above)
Comment (JakubC-projects) [reply client/src/components/timer.vue:41]: Why not? I think it's more correct to display for example 82 minutes there, than to have the clock disapear.

**Review by JakubC-projects — APPROVED** [review]

**Pull request merged**